db/blip_handler_collections.go

```diff
@@ -50,10 +50,11 @@ func (bh *blipHandler) handleGetCollections(rq *blip.Message) error {
 		// TODO: can I send the information back to the client, is this leaking data? Should I log?
 		return base.HTTPErrorf(http.StatusBadRequest, "Length of collections must match length of checkpoint ids. Request body: %v", requestBody)
 	}
+	bh.collectionMapping = requestBody.Collections
```

Comment:
Might be nice to be defensive and assert that we don't already have `bh.collectionMapping` set to ensure client's aren't sending multiple `getCollections` messages on a single connection.